### PR TITLE
Fix error with the max degree of the SRS

### PR DESCRIPTION
**Pull request by xevisalle**
Adding the blinding factors required some extra elements for the SRS: `+1` per each wire (we have 4 wires), plus `+2` for the permutation polynomial. There was a bug in the original commit, that caused errors when compiling some circuits. We fix it here by doing:

- In `srs.rs`, the `max_degree` in `setup()` is updated: `+ 6`
- In `srs.rs`, the truncated degree used in `trim()` is updated: `+ 6`.
- The `+ 6` used in `circuit.rs` has been removed as it is no longer required.
- We reverted a change in `benches/plonk.rs`, no longer required.

### CODE REVIEW

File: benches/plonk.rs [hunk 1]
```diff
@@ -85,7 +85,7 @@ fn constraint_system_benchmark(c: &mut Criterion) {
 
     let rng = &mut rand_core::OsRng;
     let label = b"dusk-network";
-    let pp = PublicParameters::setup(1 << final_degree, rng)
+    let pp = PublicParameters::setup(1 << (final_degree - 1), rng)
```
Comment (vlopes11) [inline benches/plonk.rs:88]: why is this required for a bug fix?

Comment (xevisalle) [reply benches/plonk.rs:88]: We are reverting it, this is the original code, that we had to modify in https://github.com/dusk-network/plonk/commit/8efda59b0c07821c4a1ee66f7172143326b28b48 to avoid a crash, related with the same.

File: src/commitment_scheme/kzg10/srs.rs [hunk 2]
```diff
@@ -38,20 +38,29 @@ impl PublicParameters {
         &self.opening_key
     }
 
+    /// The maximum degree is the degree of the constraint system + 6,
+    /// because adding the blinding factors requires some extra elements
+    /// for the SRS: +1 per each wire (we have 4 wires), plus +2 for the
+    /// permutation polynomial
+    const MAX_DEGREE_UPDATE: usize = 6;
```
[the remark this answers is not in the document]
Comment (xevisalle) [reply src/commitment_scheme/kzg10/srs.rs:45]: As you prefer!

File: benches/plonk.rs (hunk 1 above)
Comment (xevisalle) [reply benches/plonk.rs:88]: Because you are never using so many public parameters, as per how the benchmark loop iterating over all the possible circuit degrees was implemented. The maximum circuit degree benchmark is final degree - 1.

Comment (moCello) [reply benches/plonk.rs:88]: care to write it as a comment ;)

File: src/commitment_scheme/kzg10/srs.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (moCello) [reply src/commitment_scheme/kzg10/srs.rs:45]: Sort of curious on your and @vlopes11 opinions here

[the remark this answers is not in the document]
Comment (xevisalle) [reply src/commitment_scheme/kzg10/srs.rs:45]: ok I put this then!

**Review by moCello — APPROVED** [review]
LGTM
